doc/source/serve/production.md

```diff
@@ -583,3 +565,217 @@ This section summarizes the best practices when deploying to production:
+
+#### Example: Serve config update
+
+In the `FruitStand` example above, let's change the price of a mango in the Serve config to 4:
+
+```console
+  - name: MangoStand
+    numReplicas: 1
+    userConfig: |
+      price: 4
+```
+
+Now to update the application we apply the modified manifest:
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+$ kubectl describe
+...
+  serveDeploymentStatuses:
+  - healthLastUpdateTime: "2022-07-18T21:51:37Z"
+    lastUpdateTime: "2022-07-18T21:51:41Z"
+    name: MangoStand
+    status: UPDATING
+...
+```
+
+If we query the application, we can see that we now get a different result reflecting the updated price:
+
+```console
+$ curl -X POST -H 'Content-Type: application/json' localhost:8000 -d '["MANGO", 2]'
+8
+```
+
+#### Updating the RayCluster config
+
+The process of updating the RayCluster config is the same as updating the Serve config.
+For example, we can update the number of worker nodes to 2 in the manifest:
+
+```console
+workerGroupSpecs:
+  # the pod replicas in this group typed worker
+  - replicas: 2
+```
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+$ kubectl describe
+...
+  pendingServiceStatus:
+    appStatus: {}
+    dashboardStatus:
+      healthLastUpdateTime: "2022-07-18T21:54:53Z"
+      lastUpdateTime: "2022-07-18T21:54:54Z"
+    rayClusterName: rayservice-sample-raycluster-bshfr
+    rayClusterStatus: {}
+...
+```
+
+In the status, you can see that the `RayService` is preparing a pending cluster.
+After the pending cluster is healthy, it is switched to the active cluster and the previous cluster is terminated.
+
+## Failure Recovery
+
+Ray Serve is resilient to any component failures within the Ray cluster out of the box.
+You can read more about how worker node and process failures are handled in: {ref}`serve-ft-detail`.
+
+### Head node failures
+
+By default the Ray head node is a single point of failure: if it crashes, the entire cluster crashes and needs to be restarted.
+When running on Kubernetes, the `RayService` controller health checks the cluster and restarts it if this occurs, but this still results in some downtime.
+In Ray 2.0, `KubeRay` has added experimental support for [GCS fault tolerance](https://ray-project.github.io/kuberay/guidance/gcs-ha/#ray-gcs-ha-experimental), preventing the Ray cluster from crashing if the head node goes down.
+While the head node is recovering, Serve applications can still handle traffic but cannot be updated or recover from other failures (e.g., actors or worker nodes crashing).
+Once the GCS is recovered, the cluster will return to normal behavior.
```

Comment:
Can we tell users how to check their head node's state during runtime?

Reply:
I don't think this is the right place for that, it should be covered in kuberay/cluster management docs

Reply:
That sounds good